src/goto-instrument/contracts/contract_wrangler.cpp

```diff
@@ -0,0 +1,391 @@
+/*******************************************************************\
+
+Module: Parse and annotate contracts from configuration files
+
+Author: Qinheping Hu
+
+Date: June 2023
+
+\*******************************************************************/
+
+/// \file
+/// Parse and annotate contracts
+
+#include "contract_wrangler.h"
+
+#include <util/expr_iterator.h>
+#include <util/format_expr.h>
+#include <util/pointer_expr.h>
+#include <util/simplify_expr.h>
+#include <util/std_types.h>
+
+#include <goto-programs/remove_unused_functions.h>
+
+#include <ansi-c/ansi_c_parser.h>
+#include <ansi-c/ansi_c_typecheck.h>
+#include <goto-instrument/contracts/utils.h>
+
+contract_wranglert::contract_wranglert(
+  goto_modelt &goto_model,
+  const std::string &file_name,
+  messaget &log)
```

Comment:
I'd suggest to pass a `message_handlert` and construct the `messaget` only where and when needed.